src/coreclr/jit/optimizer.cpp

```diff
@@ -3336,60 +3208,61 @@ weight_t Compiler::optEstimateEdgeLikelihood(BasicBlock* from, BasicBlock* to, b
 }
 
 //-----------------------------------------------------------------------------
-// optSetExitWeight: Set the weight of a newly created exit, after it
-// has been added to the flowgraph.
+// optSetWeightForPreheaderOrExit: Set the weight of a newly created preheader
+// or exit, after it has been added to the flowgraph.
 //
 // Parameters:
-//   loop      - The loop
-//   preheader - The new exit block
+//   loop  - The loop
+//   block - The new preheader or exit block
 //
-void Compiler::optSetExitWeight(FlowGraphNaturalLoop* loop, BasicBlock* exit)
+void Compiler::optSetWeightForPreheaderOrExit(FlowGraphNaturalLoop* loop, BasicBlock* block)
 {
     bool hasProfWeight = true;
 
-    // Inherit first estimate from the exit target; optEstimateEdgeLikelihood
+    assert(block->GetUniqueSucc() != nullptr);
+    // Inherit first estimate from the target target; optEstimateEdgeLikelihood
```

Comment:
```suggestion
    // Inherit first estimate from the target; optEstimateEdgeLikelihood
```

Reply:
Will include this as part of a future PR.